tracing/src/span.rs

```diff
@@ -720,10 +720,61 @@ impl Span {
     ///
     /// If this span is disabled, or the resulting follows-from relationship
     /// would be invalid, this function will do nothing.
-    pub fn follows_from(&self, from: impl for<'a> Into<Option<&'a Id>>) -> &Self {
+    ///
+    /// # Examples
+    ///
+    /// Setting a `follows_from` relationship with a `Span`:
+    /// ```
+    /// # use tracing::{span, Id, Level, Span};
+    /// # fn main() {
+    /// let span1 = span!(Level::INFO, "span_1");
+    /// let span2 = span!(Level::DEBUG, "span_2");
+    /// span2.follows_from(span1);
+    /// # }
+    /// ```
+    ///
+    /// Setting a `follows_from` relationship with a `Span::current()`:
+    /// ```
+    /// # use tracing::{span, Id, Level, Span};
+    /// # fn main() {
+    /// let span = span!(Level::INFO, "hello!");
+    /// span.follows_from(Span::current());
+    /// # }
+    /// ```
+    ///
+    /// Setting a `follows_from` relationship with a `Span` reference:
+    /// ```
+    /// # use tracing::{span, Id, Level, Span};
+    /// # fn main() {
+    /// let span = span!(Level::INFO, "hello!");
+    /// let curr = Span::current();
+    /// span.follows_from(&curr);
+    /// # }
+    /// ```
+    ///
+    /// Setting a `follows_from` relationship with an `Id`:
+    /// ```
+    /// # use tracing::{span, Id, Level, Span};
+    /// # fn main() {
+    /// let span = span!(Level::INFO, "hello!");
+    /// let id = span.id();
+    /// span.follows_from(id);
+    /// # }
+    /// ```
+    ///
+    /// `&Id` will not compile.
```

Comment:
We should probably add an `Into<Option<Id>>` for `&Id` in `tracing-core`, but we should make that change separately. Let's remove this doctest, since adding that impl would break it.

Reply:
Why not keep this doctest so that when we _do_ introduce support for `Into<Option<Id>>` for `&Id` in `tracing-core`, this breaks as a reminder to fix and update docs?

Reply:
I don't think we should make an additive change in `tracing-core` break `tracing`, especially since we should try to _avoid_ bumping `tracing`'s minimum `tracing-core` dependency whenever possible?

Reply:
That's true. Removed.